saleor/payment/gateways/stripe/__init__.py

```diff
@@ -174,50 +134,49 @@ def _get_client(**connection_params):
 
 
 def _get_stripe_charge_payload(
-        payment: Payment, amount: Decimal, payment_token: str,
-        should_capture: bool):
-    shipping_address = payment.order.shipping_address
+        payment_information: Dict, should_capture: bool):
+    shipping = payment_information['shipping']
 
     # Get currency
-    currency = get_currency_for_stripe(payment.currency)
+    currency = get_currency_for_stripe(payment_information['currency'])
 
     # Get appropriate amount for stripe
-    stripe_amount = get_amount_for_stripe(amount, currency)
+    stripe_amount = get_amount_for_stripe(
+        payment_information['amount'], currency)
 
     # Get billing name from payment
-    name = get_payment_billing_fullname(payment)
+    name = get_payment_billing_fullname(payment_information)
 
     # Construct the charge payload from the data
     charge_payload = {
         'capture': should_capture,
         'amount': stripe_amount,
         'currency': currency,
-        'source': payment_token,
+        'source': payment_information['token'],
         'description': name}
 
-    if shipping_address:
+    if shipping:
         # Update shipping address to prevent fraud in Stripe
         charge_payload['shipping'] = {
             'name': name,
-            'address': shipping_address_to_stripe_dict(shipping_address)}
+            'address': shipping_to_stripe_dict(shipping)}
 
     return charge_payload
 
 
-def _create_stripe_charge(
-        client, payment, amount: Decimal, payment_token: str,
-        should_capture: bool):
+def _create_stripe_charge(client, payment_information, should_capture: bool):
     """Create a charge with specific amount, ignoring payment's total."""
     charge_payload = _get_stripe_charge_payload(
-        payment, amount, payment_token, should_capture)
+        payment_information, should_capture)
     return client.Charge.create(**charge_payload)
 
 
-def _create_transaction(payment, amount, kind, response):
+def _create_response(payment_information, kind, response, error):
     # Get currency from response or payment
     currency = get_currency_from_stripe(
-        response.get('currency', payment.currency))
+        response.get('currency', payment_information['currency']))
 
+    amount = payment_information.get('amount')
```

Comment:
Same here, as the gateway provides the default amount and currency, why don't we make the amount and currency as required fields in ```validate_gateway_response``` and then then there is no need to provide the default values in ```create_transaction```.